clang-tools-extra/clang-tidy/custom/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+set(LLVM_LINK_COMPONENTS
+  support
+  )
+
+add_clang_library(clangTidyCustomModule STATIC
+  CustomTidyModule.cpp
+  QueryCheck.cpp
+
+  LINK_LIBS
+  clangTidy
+  clangTidyUtils
+
+  DEPENDS
+  ClangDriverOptions
+  )
+
+clang_target_link_libraries(clangTidyCustomModule
+  PRIVATE
+  clangQuery
```

Comment:
Now clang-tidy is made to be dependent on clangQuery, which is a major change. Is this OK? How are build times affected?

It would be great to have @AaronBallman's opinion on this one :)

Reply:
Probably would be fine, would be good to have cmake options, to enable custom checks, and then if not enabled then support for them woudn't be included in binary

Reply:
Nice, something similar to what's done with the static-analyzer checks

Reply:
I think the suggestion from @PiotrZSL is a reasonable compromise.

I'm presuming it's not an issue that the `clangQuery` library links against a bunch of clang libraries that clang-tidy already links against, but it would be good to double-check that we're not accidentally getting two copies of those libraries in a static link build.